### PR TITLE
Added Feature to adjust data label position.

### PR DESCRIPTION
**Pull request by hb2708**
Added one variable for offset and calculated radius depending on the given  to achieve the feature referenced in  #1687   Attaching the sample result.
The value is set as follows 
`pieChart.dataLabelOffset = 0.75`
screen shot 2016-10-25 at 11 17 10 am

### CODE REVIEW

**Comment by hb2708** [comment]
Why it says All checks have failed?

**Comment by hb2708** [comment]
is anyone there?

**Comment by petester42** [comment]
The tests sometimes fail if downloading realm takes too long. But you're all good now

**Comment by hb2708** [comment]
Actually i have made one mistake in code also i corrected and pushed and it's all ok now.

So when should i expect it to get merged in master ?

**Comment by hb2708** [comment]
Hii any progress on this

**Comment by liuxuan30** [comment]
please wait - usually new feature will be reviewed by @danielgindi but he seems busy recently.

**Comment by hb2708** [comment]
is anyone looking into this?

**Comment by petester42** [comment]
Not right now. But keep in mind that since this library has to be API compatible with the android version it may take some time for your change to make it into a release.

**Comment by hb2708** [comment]
But it's very small changes which comes with very important functionality.

**Comment by hb2708** [comment]
@petester42 @liuxuan30 @danielgindi 

Any updates on this.

Is it valid or should i cancel the pull request.

**Comment by liuxuan30** [comment]
This is a customization for pie chart value labels, so I am fine with this PR, code seems fine.
However, usually it's @danielgindi 's call to decide whether to merge and port to android.

**Comment by hb2708** [comment]
@liuxuan30  that's great to here that everything is as expected. Now have to wait @danielgindi for the approval

**Comment by liuxuan30** [comment]
@hb2708 maybe longer than you expect, but if we simply merge this, people may not realize this or causing conflicts. there are other things need to be considered such as porting to android, and if it's possible to expand to other chart types, like radar chart, though this is just a clear shot.

**Comment by hb2708** [comment]
Hello Any updates on this?

**Comment by jjatie** [comment]
@liuxuan30 can we consider this for 3.1?

File: Source/Charts/Charts/PieChartView.swift [hunk 1]
```diff
@@ -71,6 +71,9 @@ open class PieChartView: PieRadarChartViewBase
     
     /// maximum angle for this pie
     fileprivate var _maxAngle: CGFloat = 360.0
+    
+    /// Extra offset value to be added in the radius to move data label outside the chart
+    fileprivate var _dataLabelOffset: CGFloat = 0.0
```
Comment (liuxuan30) [inline Source/Charts/Charts/PieChartView.swift]: maybe we need a more precise word to describe this `offset` as this is eventually a multiplier, not plus

File: Source/Charts/Charts/PieChartView.swift [hunk 2]
```diff
@@ -659,4 +662,28 @@ open class PieChartView: PieRadarChartViewBase
...
+    {
+        get
+        {
+            return _dataLabelOffset
+        }
+        set
+        {
+            _dataLabelOffset = newValue
+            
+            if _dataLabelOffset > 1.0
```
Comment (liuxuan30) [inline Source/Charts/Charts/PieChartView.swift]: any reason to lock at 1.0? how about 2.0?

Comment (jjatie) [reply Source/Charts/Charts/PieChartView.swift]: Or 1000?

**Comment by jjatie** [comment]
@liuxuan30 if this isn't resolved in time for 3.1, let's just push it to 4.0.0

File: Source/Charts/Charts/PieChartView.swift (hunk 2 above)
Comment (liuxuan30) [reply Source/Charts/Charts/PieChartView.swift]: it's a multiplier, I guess we need to let the consumer knows this so they won't write 1000x..

Comment (jjatie) [reply Source/Charts/Charts/PieChartView.swift]: I don't think it should be though. It seems very unintuitive.

Comment (hb2708) [reply Source/Charts/Charts/PieChartView.swift]: 1.0 is max.
like 0 to 100 i.e 0.0 to 1.0

File: Source/Charts/Charts/PieChartView.swift (hunk 1 above)
Comment (hb2708) [reply Source/Charts/Charts/PieChartView.swift]: Yes i agree

**Comment by hb2708** [comment]
Is all good with the change? Or improvements required?

**Comment by liuxuan30** [comment]
hello, you didn't address the review comment we have; and yet it has conflicts

**Comment by hb2708** [comment]
I add comments,

and the conflict was not there when i created PR

**Pull request closed without merging**